pkg/api/defaults.go

```diff
@@ -714,7 +714,26 @@ func (p *Properties) setWindowsProfileDefaults(isUpgrade, isScale bool) {
 				windowsProfile.ImageVersion = "latest"
 			}
 		}
+	} else if isUpgrade {
+		// Image reference publisher and offer only can be set when you create the scale set so we keep the old values.
+		// Reference: https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-upgrade-scale-set#create-time-properties
+		if windowsProfile.WindowsPublisher == AKSWindowsServer2019OSImageConfig.ImagePublisher && windowsProfile.WindowsOffer == AKSWindowsServer2019OSImageConfig.ImageOffer {
+			if windowsProfile.ImageVersion == "" {
+				windowsProfile.ImageVersion = AKSWindowsServer2019OSImageConfig.ImageVersion
+			}
+			if windowsProfile.WindowsSku == "" {
+				windowsProfile.WindowsSku = AKSWindowsServer2019OSImageConfig.ImageSku
+			}
```

Comment:
Just curious: does this mean current windows upgrade is not working, or someone is picking up the value for us?

Reply:
Current RP hardcoded a version so it will always set `ImageVersion ` and `WindowsSku` so they are always not empty.

Reply:
This will block user upgrade their images even in upgrade operation. Does this version of AKS-E guarantees to support all the history published windows images?

Reply:
@gtxistxgao I cannot fully understand your question. This change only impacts those users who have created a cluster with `AKSWindowsServer2019OSImageConfig` or `WindowsServer2019OSImageConfig`. And aks-engine only fills `ImageVersion` and `WindowsSku` when they are not empty.
aks-engine only sets the latest default `ImageVersion` and `ImageSku`. This is similar for creating logic.

Reply:
I see. My misunderstanding. This echos the contract to use tested image version.

Reply:
So this is really making the windows upgrade with the latest AKS-Engine supported image. Thanks for the info @AbelHu @gtxistxgao